CRM/Contribute/BAO/Contribution.php

```diff
@@ -5669,4 +5674,79 @@ public static function replaceContributionTokens(
     return $contributionDetails;
   }
 
+  /**
+   * Record adjusted amount.
+   *
+   * @param int $updatedAmount
+   * @param int $paidAmount
+   * @param int $contributionId
+   *
+   * @param int $taxAmount
+   * @param bool $updateAmountLevel
+   *
+   * @return bool|\CRM_Core_BAO_FinancialTrxn
+   */
+  public static function recordAdjustedAmt($updatedAmount, $paidAmount, $contributionId, $taxAmount = NULL, $updateAmountLevel = NULL) {
```

Comment:
@JoeMurray actually I have only moved the ```CRM_Event_BAO_Participant::recordAdjustAmt(..)``` to this class. The function body isn't changed at all which is used to record financial trxn on the adjustment made upon event's fee change and now membership type change too, via backoffice form. Also ensured that the correct tax amount is recorded in DB.In fact. Change in the price of related line item(s) is handled in another BAO function ```changeFeeSelction(..)``` where this fn is called, please check below in LineItem BAO file.  

 I am afraid I cannot change the accessibility of this function since it's been called by LineItem BAO function [here](https://github.com/civicrm/civicrm-core/pull/10370/files#diff-e57ed384488b5985e5a0a5c166a6b0f0R816)